ping-payload-extensions/extensions/template.md

```diff
@@ -0,0 +1,25 @@
+# [Title]
+
+[Description]
+
+## version [Version Number]
+
+[Version Description]
+
+Ping payload
+```python
+[Container Name] = Container(
+  type: [Type Number],
+  version: [Type Version],
+  payload: Container([Payload Description])
+)
+```
+
+Pong payload
+```python
+[Container Name] = Container(
+  type: [Type Number],
+  version: [Type Version],
+  payload: Container([Payload Description])
```

Comment:
I think the only one of those reasons that I'm compelled by is that unions are akward to implement any kind of rolling upgrade with. Given that, I think I'd advocate for a single `type` field that is something like `u16` and a `ByteList` payload that is the encoded payload.

Reply:
Said explicitely, it would be

```python
Container(type: u16, payload: ByteList[max_length=XXXX])
```

where clients implement application specific logic for decoding the `payload` value into some number of known SSZ types that are supported in that network.

Reply:
Yes, this is also what I mentioned as 1 of the two options here: https://github.com/ethereum/portal-network-specs/pull/348#discussion_r1810278302

And I agree that type + version is quite overkill, especially with the amount of bytes reserved for each.

Reply:
This sounds like the play and is aligned with what Kim said. I think a u32 would still be very nice. I might be being a little hyperbolic, but I wouldn't want to deal with the possibility of running out of types. So it will probably be fine with a u16, but if we did run out that would be a dreadful experience to deal with. But it is probably fine

Reply:
For errors I think a u16 is more then enough to be clear*